File: cameraview/src/main/java/com/base/cameraview/picture/FullPictureRecorder.java
package com.base.cameraview.picture;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.base.cameraview.CameraLogger;
import com.base.cameraview.PictureResult;

/**
 * Helps with logging.
 */
public abstract class FullPictureRecorder extends PictureRecorder {
    private static final String TAG = FullPictureRecorder.class.getSimpleName();
    protected static final CameraLogger LOG = CameraLogger.create(TAG);

    public FullPictureRecorder(@NonNull PictureResult.Stub stub,
                               @Nullable PictureResultListener listener) {
        super(stub, listener);
    }
}

File: cameraview/src/main/java/com/base/cameraview/picture/FullPictureRecorder.kt
package com.base.cameraview.picture

import com.base.cameraview.CameraLogger
import com.base.cameraview.PictureResult

/**
 * Helps with logging.
 */
abstract class FullPictureRecorder(
    stub: PictureResult.Stub,
    listener: PictureResultListener?
) : PictureRecorder(stub, listener) {
    companion object {
        val TAG: String = FullPictureRecorder::class.java.simpleName
        val LOG: CameraLogger = CameraLogger.create(TAG)
    }
}
